static/js/stream_popover.js

```diff
@@ -436,24 +440,47 @@ function build_move_topic_to_stream_popover(current_stream_id, topic_name) {
         send_notification_to_old_thread = send_notification_to_old_thread === "on";
         current_stream_id = Number.parseInt(current_stream_id, 10);
 
+        if (old_topic_name.trim() === new_topic_name.trim()) {
+            // We use `undefined` to tell the server that
+            // there has been no change in the topic name.
+            new_topic_name = undefined;
+        }
+
+        if (select_stream_id === current_stream_id) {
+            // We use `undefined` to tell the server that
+            // there has been no change in stream.
+            select_stream_id = undefined;
+        }
+
+        let propagate_mode = "change_all";
+        if (message !== undefined) {
+            // We already have the message_id here which means that modal is opened using
+            // message popover.
+            propagate_mode = $("#move_topic_modal select.message_edit_topic_propagate").val();
+            message_edit.move_topic_containing_message_to_stream(
+                message.id,
+                select_stream_id,
+                new_topic_name,
+                send_notification_to_new_thread,
+                send_notification_to_old_thread,
+                propagate_mode,
+            );
+            return;
+        }
+
         dialog_widget.show_dialog_spinner();
         message_edit.with_first_message_id(
             current_stream_id,
             old_topic_name,
             (message_id) => {
-                if (old_topic_name.trim() === new_topic_name.trim()) {
-                    // We use `undefined` to tell the server that
-                    // there has been no change in the topic name.
-                    new_topic_name = undefined;
-                }
-
                 if (old_topic_name && select_stream_id) {
```

Comment:
I have not added these conditions above in the `message !== undefined` block, as I am not sure whether it is required.
`old_topic_name` should always be non-null string. And for `select_stream_id` also, there is no reset button for this widget.

Reply:
I'd look at the history for why this code is there; one possibility is that `new_topic-name` is what was meant, since that's the parameter used in the block inside this conditional? I'm not sure I understand.